keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/UnifiedMCMCTest.java

```diff
@@ -0,0 +1,49 @@
+package io.improbable.keanu.algorithms.mcmc;
+
+import com.google.common.collect.ImmutableList;
+import io.improbable.keanu.algorithms.PosteriorSamplingAlgorithm;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import org.junit.Test;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class UnifiedMCMCTest {
+
+    @Test
+    public void checksIfNetworkIsDifferentiable() {
+        BayesianNetwork network = mock(BayesianNetwork.class);
+        Vertex vertex = new GaussianVertex(0, 1);
+        when(network.getTopLevelLatentVertices()).thenReturn(ImmutableList.of(vertex));
+        when(network.getLatentOrObservedVertices()).thenReturn(ImmutableList.of(vertex));
+        when(network.getContinuousLatentVertices()).thenReturn(ImmutableList.of(vertex));
+
+        MCMC.withDefaultConfig().forNetwork(network);
+        verify(network).getNonDifferentiableVertices();
```

Comment:
Oh I see why you need the mock. Is it a requirement that the unit under test calls this method? Or is it's only requirement that it returns the appropriate sampling algorithm?

Reply:
I guess the latter, will remove that particular test.